proof/refine/AARCH64/Interrupt_R.thy

```diff
@@ -971,52 +971,54 @@ lemma handle_reserved_irq_corres[corres]:
      apply (rule corres_split)
-        apply (rule corres_machine_op, rule corres_eq_trivial ; (simp add: dc_def no_fail_bind)+)+
+        apply (rule corres_machine_op, rule corres_eq_trivial; simp)+
       apply ((wp | simp)+)[4]
 
   apply (rule corres_gen_asm2)
   apply (case_tac st, simp_all add: irq_state_relation_def split: irqstate.split_asm)
-   apply (simp add: getSlotCap_def bind_assoc)
+    apply (simp add: getSlotCap_def bind_assoc)
+    apply (rule corres_guard_imp)
+      apply (rule corres_split[OF getIRQSlot_corres])
+        apply simp
+        apply (rule corres_split[OF get_cap_corres,
+                                  where R="\<lambda>rv. einvs and valid_cap rv"
+                                   and R'="\<lambda>rv. invs' and valid_cap' (cteCap rv)"])
+          apply (rule corres_underlying_split[where r'=dc])
+             apply (case_tac xb, simp_all add: doMachineOp_return)[1]
+              apply (clarsimp simp add: when_def doMachineOp_return)
+              apply (rule corres_guard_imp, rule sendSignal_corres)
+               apply (clarsimp simp: valid_cap_def valid_cap'_def arch_mask_irq_signal_def
+                                     maskIrqSignal_def do_machine_op_bind doMachineOp_bind)+
+            apply corres
+              apply (rule corres_machine_op, rule corres_eq_trivial; simp)+
+             apply wpsimp+
+    apply fastforce
    apply (rule corres_guard_imp)
-     apply (rule corres_split[OF getIRQSlot_corres])
-       apply simp
-       apply (rule corres_split[OF get_cap_corres,
-                                 where R="\<lambda>rv. einvs and valid_cap rv"
-                                  and R'="\<lambda>rv. invs' and valid_cap' (cteCap rv)"])
-         apply (rule corres_underlying_split[where r'=dc])
-            apply (case_tac xb, simp_all add: doMachineOp_return)[1]
-             apply (clarsimp simp add: when_def doMachineOp_return)
-             apply (rule corres_guard_imp, rule sendSignal_corres)
-              apply (clarsimp simp: valid_cap_def valid_cap'_def arch_mask_irq_signal_def
-                                    maskIrqSignal_def do_machine_op_bind doMachineOp_bind)+
-           apply corres
-             apply (rule corres_machine_op, rule corres_eq_trivial; simp)+
-            apply wpsimp+
-   apply fastforce
-  apply (rule corres_guard_imp)
-    apply (rule corres_split)
-       apply simp
-       apply (rule corres_split[OF timerTick_corres corres_machine_op])
-         apply (rule corres_eq_trivial, wpsimp+)
-      apply (rule corres_machine_op)
-      apply (rule corres_eq_trivial; simp)
-     apply wp+
-   apply (clarsimp simp: invs_distinct invs_psp_aligned schact_is_rct_def)
-  apply clarsimp
+     apply (rule corres_split)
+        apply simp
+        apply (rule corres_split[OF timerTick_corres corres_machine_op])
+          apply (rule corres_eq_trivial, wpsimp+)
+       apply (rule corres_machine_op)
+       apply (rule corres_eq_trivial; simp)
```

Comment:
It's possible that you can replace all that with just `apply (corres corres: timerTick_corres corres_machine_op)`. Worth a shot if you have time.